nextcord/message.py

```diff
@@ -543,6 +544,62 @@ def flatten_handlers(cls):
     return cls
 
 
+class _MessageInteractionOptional(TypedDict, total=False):
+    member: Member
+
+
+class MessageInteractionPayload(_MessageInteractionOptional):
+    id: Snowflake
+    type: InteractionType
+    name: str
+    user: User
+
+
+class MessageInteraction:
+    """Represents a message's interaction data, regardless of application.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    data: :class:`dict`
+        The raw data from the interaction.
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The application ID that the interaction was for.
+    user: :class:`User`
+        The user or that sent the interaction;
+        Or member if the interaction occured in a guild.
+    """
+
+    __slots__ = (
+        "_state",
+        "data",
+        "id",
+        "type",
+        "name",
+        "user",
+    )
+
+    def __init__(self, *, data: MessageInteractionPayload, guild: Guild, state: ConnectionState):
```

Comment:
Still needs to be optional
```suggestion
    def __init__(self, *, data: MessageInteractionPayload, guild: Optional[Guild], state: ConnectionState):
```